File: Python/Stack/Q3_1.py
from BaseImplementation import Stack


def array_to_three_stacks(arr):
    maxl = len(arr)
    # create 3 stacks
    fs = Stack()
    ss = Stack()
    ts = Stack()

    #create 3 arrays
    fa = arr[0: maxl / 3]
    sa = arr[maxl / 3: (2 * maxl) / 3]
    ta = arr[(2 * maxl) / 3: maxl]

    for i in fa:
        fs.push(i)
    for j in sa:
        ss.push(j)
    for k in ta:
        ts.push(k)

    return fs, ss, ts


arr_test = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]

a = array_to_three_stacks(arr_test)
print a[0].peek()
print a[1].peek()
print a[2].peek()

File: Java/Hired Programming Practice/src/StacksAndQueues/Q3_1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package StacksAndQueues;

import java.util.Arrays;

/**
 *
 * @author marcsantiago
 */
//Write a method that converts an array into three stacks
public class Q3_1 {
   
    public BaseStackImplementation [] arrayToThreeStacks(Object [] list){
        
        int max = list.length;
        //create stacks
        BaseStackImplementation firstStack = new BaseStackImplementation();
        BaseStackImplementation secondStack = new BaseStackImplementation();
        BaseStackImplementation thirdStack = new BaseStackImplementation();
        //create three list
        Object [] first = Arrays.copyOfRange(list, 0, max / 3);
        Object [] second = Arrays.copyOfRange(list, max / 3, (2 * max) / 3);
        Object [] third  = Arrays.copyOfRange(list,(2 * max) / 3, max);
        
        
        for(Object o: first){
            firstStack.push(o);
        }
        for(Object o: second){
            secondStack.push(o);
        }
        for(Object o: third){
            thirdStack.push(o);
        }
        
        BaseStackImplementation [] all = {firstStack, secondStack, thirdStack};
        
        return all;
    }     
}

